src/compilation/compiler.rs

```diff
@@ -162,6 +193,14 @@ impl Compiler {
         }
     }
 
+    /// This function compiles the provided program, so that its LLVM can be [optimized][a],
+    /// [printed][b], or [written to a file][c].
```

Comment:
```suggestion
    /// This function compiles the provided program. Once the program is compiled, its LLVM IR can be [optimized][a],
    /// [printed][b], or [written to a file][c].
```

Reply:
Agreed! I trimmed it a bit so that it all fits in one sentence (since the top-level doc should be one sentence I think)
```
    /// This function compiles the provided program; once compiled, its LLVM IR can be [optimized][a],
    /// [printed][b], or [written to a file][c].
```